go/ocr2/decryptionplugin/decryption.go

```diff
@@ -41,10 +41,16 @@ func (f DecryptionReportingPluginFactory) NewReportingPlugin(rpConfig types.Repo
 		})
 		return nil, types.ReportingPluginInfo{}, fmt.Errorf("unable to decode reporting plugin config: %w", err)
 	}
+	if int(pluginConfig.Config.K) <= rpConfig.F || int(pluginConfig.Config.K) > 2*rpConfig.F+1 {
+		f.Logger.Error("invalid configuration: decryption threshold K must satisfy F < K <= 2F+1", commontypes.LogFields{
+			"configDigest": rpConfig.ConfigDigest.String(),
+		})
+		return nil, types.ReportingPluginInfo{}, fmt.Errorf("invalid configuration: decryption threshold K must satisfy F < K <= 2F+1")
+	}
 
 	info := types.ReportingPluginInfo{
 		Name:          "ThresholdDecryption",
-		UniqueReports: false, // Aggregating any f+1 valid decryption shares result in the same plaintext. Must match setting in OCR2Base.sol.
+		UniqueReports: false, // Aggregating any k valid decryption shares result in the same plaintext. Must match setting in OCR2Base.sol.
```

Comment:
Is this comment correct? Doesn't it only mean that OCR will provide 2F+1 observations, regardless of K?

Reply:
OCR2 will provide 2F+1 _observations_ anyway.
Setting this flag to false means that the leader is only required to gather F+1 _reports_.
This means that the leader can equivocate and create _final reports_ with different sets of 2F+1 observations.
However in this plugin this is ok, as any K observations will result in the same plaintext, hence any final report will have the same plaintext.
Does it make sense now?

Reply:
Ah right, it's about reports, thanks!